src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -1583,9 +1634,13 @@ private static DeclarationModifiers GetAllowedModifiers(SyntaxKind kind)
                 case SyntaxKind.RemoveAccessorDeclaration:
                     return s_accessorModifiers;
 
-                case SyntaxKind.EnumMemberDeclaration:
-                case SyntaxKind.Parameter:
                 case SyntaxKind.LocalDeclarationStatement:
+                    return s_localDeclarationModifiers;
```

Comment:
I believe this change should be reverted - in fact I wouldn't even list parameter, local declaration in this switch and just let it fall through to default case.

Reply:
i personally disagree.  SyntaxGenerator attempts to be a 'smoothing' out API, where one can operate on any nodes where a *sensible* interpretation can be provided.  In this case, Locals can def have modifiers, so it seems reasonable to support this.

I'd def like to hear @mattwar 's thoughts though as maybe i'm entirely offbase and he strongly felt these shoudl be out of scope of SyntaxGenerator.